### ISSUE TITLE
Empty list when deserializing namespaced XML with children that are not namespaced

### ISSUE THREAD

**Issue by bertterheide** [schmittjoh/serializer#850]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | no

When deserializing XML into an object of which the root is namespaced but the a underlying list is not, the resulting object's property will contain an empty list.

## Steps required to reproduce the problem

1. Deserialize XML of which the root is namespaced but a underlying list is not into an object;
2. Inspect the list

## Expected Result
* The underlying list containing elements

## Actual Result
* The underlying list is empty

**Comment by bertterheide** [comment]
I have provided a fix for this issue in #851.

**Comment by goetas** [comment]
not sure if this is a bugfix... 

if you add : 

```php

class PersonCollection
{
    /**
     * @Type("ArrayCollection<JMS\Serializer\Tests\Fixtures\Person>")
     * @XmlList(entry = "person", inline = true, namespace="http://example.com/namespace")
     */
    public $persons;
}
```

(just added `namespace="http://example.com/namespace` on the  `$persons` property)

everything works well.

**Comment by goetas** [comment]
I guess your code does exactly this https://github.com/schmittjoh/serializer/blob/62c7ff6d61f8692eac8be024c542b3d9d0ab8c8a/src/JMS/Serializer/XmlDeserializationVisitor.php#L282 right?

**Comment by bertterheide** [comment]
Exactly.

**Comment by bertterheide** [comment]
It should not be required to set a namespace on XmlList, and it should try to fall back to the document’s namespace.

**Comment by bertterheide** [comment]
@goetas will this issue be merged?

**Comment by goetas** [comment]
yes, looks good to me, just have to make sure there will be no bc breaks

**Comment by bertterheide** [comment]
Have you been able to check this issue for BC breaks? I need this issue fixed so I can upgrade to the latest version of the serializer.

**Comment by goetas** [comment]
fixed in #851
